### PR TITLE
[Issue 97] Sensor locations may transmit multiple sensor data types

### PR DESCRIPTION
**Pull request by carlpulley**
* [x] `SensorNet` and `SensorNetValue` are capable of holding multiple sensor data types
* [x] revised `UserExercisesClassifier` and `ExerciseModel` actor assumptions in light of these changes
* [x] updated unit testing
* [x] `sbt test`
* [x] ready for review/merge

---
This PR aims at closing issue #97

### CODE REVIEW

File: server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesClassifier.scala [hunk 1]
```diff
@@ -79,18 +79,19 @@ class UserExercisesClassifier(sessionProps: SessionProperties, modelProps: Props
...
-      val sensorMap = sdwls.sensorData.groupBy(_.location).mapValues(_.flatMap(_.data))
-      val blockSize = sensorMap(SensorDataSourceLocationWrist).length
+      // (SensorDataSourceLocation, Int) -> List[SensorData]
+      val sensorMap: Map[SensorDataSourceLocation, List[List[SensorData]]] = sdwls.sensorData.groupBy(_.location).mapValues(_.map(_.data))
+      val blockSize = sensorMap(SensorDataSourceLocationWrist).head.length
       require(
-        sensorMap.values.forall(_.length == blockSize),
-        "all sensor data locations have a common data length"
+        sensorMap.values.forall(_.forall(_.length == blockSize)),
+        "all sensor data location points have a common data length"
```
Comment (mpetruska) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesClassifier.scala]: While this is a reasonable assumption now, the number of samples can differ in the future (e.g. switching the watch to a 50 Hz sampling rate).

Comment (mpetruska) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesClassifier.scala]: Talked with Carl, re/sampling should be done.

Comment (mpetruska) [inline server/exercise/src/main/scala/com/eigengo/lift/exercise/UserExercisesClassifier.scala]: Opened issue 103 (https://github.com/eigengo/lift/issues/103) to address the possible sampling frequency difference of sensors.

**Comment by carlpulley** [comment]
buildme

**Pull request merged**